contributors/devel/code-generation-outside-kubernetes.md

```diff
@@ -0,0 +1,54 @@
+# Code generation (from the experience of developing service-catalog)
+
+## Code generator binaries
+There are several libraries in the Kubernetes tree that are responsible for
+various code generation operations. Those go based libraries are currently
+located at cmd/libs/go2idl/. In each directory, there is a main.go file
+that currently has hard coded an internal path for boilerplate.go.txt.
+However, there now exists a repo-infra repository that can be used instead.
+These files serve mostly as wrappers for executing and utilizing gengo.
+
+## Code generation arguments
+In Kubernetes, most of the code generation execution handling is in the
+Makefile.generated_files and is in the .generated_files target. The rule
+complexity is due to Kubernetes generating a DAG in order to determine
+which files need rebuilt and code regenerated for.
```

Comment:
Fixed verb. I wish I understood all the makefile dependency generation better...